Mathlib/MeasureTheory/Function/UnifTight.lean

```diff
@@ -0,0 +1,511 @@
+/-
+Copyright (c) 2023 Igor Khavkine. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Igor Khavkine
+-/
+import Mathlib.Order.Filter.IndicatorFunction
+import Mathlib.MeasureTheory.Function.ConvergenceInMeasure
+import Mathlib.MeasureTheory.Function.L1Space
+import Mathlib.MeasureTheory.Function.UniformIntegrable
+
+/-!
+# Uniform tightness
+
+This file contains the definitions for uniform tightness for a family of Lp functions.
+It is used as a hypothesis to the version of Vitali's convergence theorem for Lp spaces
+that works also for spaces of infinite measure. This version of Vitali's theorem
+is also proved later in the file.
+
+## Main definitions
+
+* `MeasureTheory.UnifTight`:
+  A sequence of functions `f` is uniformly tight in `L^p` if for all `ε > 0`, there
+  exists some measurable set `s` with finite measure such that the Lp-norm of
+  `f i` restricted to `sᶜ` is smaller than `ε` for all `i`.
+
+# Main results
+
+* `MeasureTheory.unifTight_finite`: a finite sequence of Lp functions is uniformly
+  tight.
+* `MeasureTheory.tendsto_Lp_notFinite_of_tendsto_ae`: a sequence of Lp functions which is uniformly
+  integrable and uniformly tight converges in Lp if they converge almost everywhere.
+* `MeasureTheory.tendstoInMeasure_notFinite_iff_tendsto_Lp`: Vitali convergence theorem:
+  a sequence of Lp functions converges in Lp if and only if it is uniformly integrable,
+  uniformly tight and converges in measure.
+
+## Tags
+uniform integrable, uniformly absolutely continuous integral, Vitali convergence theorem
+-/
+
+
+namespace MeasureTheory
+
+open Classical Set Filter Topology MeasureTheory NNReal ENNReal
+
+variable {α β ι : Type*} {m : MeasurableSpace α} {μ : Measure α} [NormedAddCommGroup β]
+
+
+/- Counterpart of `tendsto_indicator_ge` from `MeasureTheory.Function.UniformIntegrable`.
+   It is used in `lintegral_indicator_compl_le`, so it is more convenient
+   to formulate it for `f` valued in `ENNReal`. Could be wrapped with `nnnorm` to make it
+   more general. -/
+theorem tendsto_ENNReal_indicator_lt (f : α → ℝ≥0∞) (x : α) :
+    Tendsto (fun M : ℕ => { x | f x < 1 / (↑M + 1) }.indicator f x) atTop (𝓝 0) := by
```

Comment:
A generalization: if `M` is a `LinearOrderedCommMonoidWithZero` (in fact, any linear order with `∀ x, 0 ≤ x` but we don't have a typeclass for this), `f : α → M`, and `g : β → M`, then `Tendsto (fun m ↦ {x | f x < m}.indicator f x) (nhds 0) (pure 0)` (and same for `≤`).

Reply:
The generalization may be true, but I'm not sure how that generalization would replace `tendsto_ENNReal_indicator_lt` in the proof of `lintegral_indicator_compl_le`. I've left things as they are for now.